ledger/src/blockstore.rs

```diff
@@ -3617,6 +3659,27 @@ impl Blockstore {
         );
         Ok(())
     }
+
+    pub fn turbine_slots_stats_inc_fec_set_index_count(&self, slot: Slot, fec_set_index: u32) {
+        let mut turbine_slots_stats = self.turbine_slots_stats.lock().unwrap();
+        let slot_stats = turbine_slots_stats.entry(slot).or_default();
+        slot_stats.inc_fec_set_index_count(fec_set_index);
+    }
+
+    pub fn turbine_slots_stats_get_min_batch_count(&self, slot: Slot) -> usize {
+        let turbine_slots_stats = self.turbine_slots_stats.lock().unwrap();
+        let slot_stats = turbine_slots_stats.get(&slot);
+        if let Some(stats) = slot_stats {
+            stats.get_min_batch_count()
+        } else {
+            0
+        }
+    }
+
+    pub fn turbine_slots_stats_remove_slot(&self, slot: Slot) -> Option<TurbineSlotStats> {
+        let mut turbine_slots_stats = self.turbine_slots_stats.lock().unwrap();
+        turbine_slots_stats.remove(&slot)
+    }
```

Comment:
I would suggest have a struct:
```rust
struct TurbineSlotStats {
    shred_counts: HashMap<
          Slot, 
          HashMap</*fec set index:*/u32, /*count:*/usize>
    >,
}
```
and encapsulate all the bookkeeping in that struct.
Right now, the logic is spread across two different structs, and feels like this is unnecessary bloating blockstore api.

Reply:
restructured